cache/lib/cache/authentication.ex

```diff
@@ -1,9 +1,11 @@
 defmodule Cache.Authentication do
   @moduledoc """
-  Authentication for CAS operations using the /api/projects endpoint.
+  Authentication for CAS operations.
 
   This module validates that a request has proper authorization to access a project
-  by calling the server's /api/projects endpoint and caching the results.
+  by first attempting to decode JWT tokens locally and extract the projects claim.
+  For non-JWT tokens (e.g., project tokens), it falls back to calling the server's
+  /api/projects endpoint and caching the results.
```

Comment:
it'd be great if the CI didn't take the auth hit – what can we do here to skip the auth call to the server for the project token? Would moving to the [account token](https://linear.app/tuist/issue/TUI-54/account-tokens), that we definitely still want to do soon-ish, make skipping the call to the server API easier?